lib/fog/opennebula/requests/compute/image_pool.rb

```diff
@@ -0,0 +1,32 @@
+module Fog
+  module Compute
+    class OpenNebula
+      class Real
+        def image_pool(filter = { })
+
+          images = ::OpenNebula::ImagePool.new(client)
+          if filter[:mine].nil?
+            images.info!
+          else
+            images.info_mine!
+          end # if filter[:mine].nil?
+
+          if not filter[:id].nil?
+            images.each do |i|
+              if filter[:id] == i.id
+                return [ i ] # return an array with only one element - found image
+              end
+            end
+          end
+          images.delete nil
+          images
+        end #def image_pool
+      end #class Real
+
+      class Mock
+        def image_pool(filter = { })
```

Comment:
That's great! I'm just wondering why, but now it makes sense! haha

On Wed, Apr 15, 2015 at 11:02 PM, EYurchenko <notifications@github.com>
wrote:

> In lib/fog/opennebula/requests/compute/image_pool.rb
> <https://github.com/fog/fog/pull/3527#discussion_r28479284>:
>
> > +          end # if filter[:mine].nil?
> > +
> > +          if not filter[:id].nil?
> > +            images.each do |i|
> > +              if filter[:id] == i.id
> > +                return [ i ] # return an array with only one element - found image
> > +              end
> > +            end
> > +          end
> > +          images.delete nil
> > +          images
> > +        end #def image_pool
> > +      end #class Real
> > +
> > +      class Mock
> > +        def image_pool(filter = { })
>
> Hi @mfbmina <https://github.com/mfbmina> , I actually copied initial code
> layout from other requests that already had these empty Mocks. Removing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fog/fog/pull/3527/files#r28479284>.
>



-- 
*Web Developer at Zertico*

*Technician in Industrial Computers by CEFET-MGGraduating in Computer
Science by UNIFEI*

*Github: https://github.com/mfbmina <https://github.com/mfbmina>Linkedin:
www.linkedin.com/profile/view?id=263114850
<http://www.linkedin.com/profile/view?id=263114850>*